Add  `orientation` parameter to Boxplot and deprecate `vert` (#28074)
matplotlib/matplotlib: doc/api/next_api_changes/deprecations/28074-TS.rst

``boxplot`` and ``bxp`` *vert* parameter
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The parameter *vert: bool* and the *boxplot.vertical* rcparam has
been deprecated on `~.Axes.boxplot` and `~.Axes.bxp`.
It is replaced by *orientation: {"vertical", "horizontal"}* for API

> consistency.

Reviewer: Please add the deprecatio of the rcParam here as well.

Author: Do you mean in the heading? or under its own heading?

Reviewer: You can do it under this heading.

> ``boxplot`` and ``bxp`` *vert* parameter, and ``boxplot.vertical`` rcparam

Reviewer: Using the actual pattern is easier to recognize:
```suggestion
``boxplot`` and ``bxp`` *vert* parameter, and ``rcParams["boxplot.vertical"]``
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The parameter *vert: bool* has been deprecated on `~.Axes.boxplot` and
`~.Axes.bxp`. It is replaced by *orientation: {"vertical", "horizontal"}*
for API consistency.

> The *boxplot.vertical* rcparam which controlled the orientation of ``boxplot``
> is deprecated as the new API does not require its use.

Reviewer: ```suggestion
``rcParams["boxplot.vertical"]``, which controlled the orientation of ``boxplot``,
is deprecated without replacement.
```
It's an idependent design decision that it's not reasonable/necessary to control the orientation via rcParams.
